routing.go

```diff
@@ -226,14 +273,12 @@ func (dht *IpfsDHT) GetValues(ctx context.Context, key string, nvals int) ([]rou
 
 	// run it!
 	_, err = query.Run(ctx, rtp)
```

Comment:
A few days ago, I understood more from your remark. I just fixed an issue where `sentRes` wasn't incremented when needed (it collects the number of results sent).

However, when there is no result, we might get an error from the lower layers, but that doesn't necessarily mean the GetValues must return an error. Instead it might mean that there is no result at all and we must not trigger an error. The previous version of the code silents some errors in this case, I see no reason to change that in this PR.